src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs

```diff
@@ -203,7 +204,7 @@ private void OnDataReceived(byte[] data, MessageType messageType)
                         _logger.LogWarning("Dropped unsolicited Stream Item message for invocation '{invocationId}'", streamItem.InvocationId);
                         return;
                     }
-                    DispatchInvocationStreamItem(streamItem, irq);
+                    DispatchInvocationStreamItemAsync(streamItem, irq);
```

Comment:
I think this introduces a race which can result in items not being reported in order. We might need a queue for this.